gh-64551: Add RFC 5034 AUTH support to poplib (#142613)
python/cpython: Doc/library/poplib.rst

   .. versionadded:: 3.4

.. method:: POP3.auth(self, mechanism, authobject, *, initial_response_ok=True)

   Authenticate using the POP3 ``AUTH`` command as specified in :rfc:`5034`.

   If *initial_response* is provided (``bytes`` or ``str``), it is

>    base64-encoded and appended to the command after a single space.

Reviewer: ```suggestion
   If *initial_response* is provided (as :class:`bytes` or :class:`str`),
   it is base64-encoded and appended to the command after a single space.
```

   If *authobject* is provided, it is called with the server’s ``bytes``
   challenge (already base64-decoded) and must return the client response

>    (``bytes`` or ``str``).  Return ``b'*'`` to abort the exchange.

Reviewer: ```suggestion
   If *authobject* is provided, it is called with the server’s ``bytes``
   challenge (already base64-decoded) and must return the client response
   as a :class:`bytes` or :class:`str`. If ``b'*'`` is returned, the exchange
   is aborted.
```

> 

Reviewer: ```suggestion
   (``bytes`` or ``str``).  Return ``b'*'`` to abort the exchange.

   .. versionadded:: next
```
